File: Zarina/TaZQL/src/ast/expression/comparison/GreaterThan.java
package ast.expression.comparison;

import ast.expression.Binary;
import ast.expression.Expression;
import ast.expression.IExpressionVisitor;
import ast.type.ChoiceType;

// >

public class GreaterThan extends Binary {
									
	public GreaterThan (Expression leftExp, Expression rightExp) {
		super(leftExp, rightExp);
	}
									
	@Override
	public <T> T accept(IExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public String toString() {
		return super.getLeftExpression().toString() + " > " + super.getRightExpression().toString();
	}
	
	@Override
	public ChoiceType getType() {
		return new ChoiceType();
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binary/GreaterThan.cs
using AST.Nodes.Interfaces;

namespace AST.Nodes.Expressions.Binary
{
    public class GreaterThan : BaseBinary
    {
        
        public GreaterThan(BaseExpression left, BaseExpression right, PositionInText position)
            : base(left, right, position)
        {}

        public override T Accept<T>(ASTVisitors.Interfaces.IExpressionVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return ">";
        }
    }
}
